e2e/e2etests/test_precompiles_distribute.go

```diff
@@ -0,0 +1,196 @@
+	utils.RequireTxSuccessful(r, receipt, "distribute should succeed when distributing within balance and allowance")
+
+	balanceShouldBe(r, 500, checkZRC20Balance(r, spenderAddress))
+	balanceShouldBe(r, 500, checkZRC20Balance(r, lockerAddress))
+	balanceShouldBe(r, 500, checkCosmosBalance(r, feeCollectorAddress, zrc20Denom))
+
+	eventDitributed, err := dstrContract.ParseDistributed(*receipt.Logs[0])
+	require.NoError(r, err)
+	require.Equal(r, zrc20Address, eventDitributed.Zrc20Token)
+	require.Equal(r, spenderAddress, eventDitributed.Zrc20Distributor)
+	require.Equal(r, fiveHundred.Uint64(), eventDitributed.Amount.Uint64())
+
+	// After one block the rewards should have been distributed and fee collector should have 0 ZRC20 balance.
+	r.WaitForBlocks(1)
+	balanceShouldBe(r, 0, checkCosmosBalance(r, feeCollectorAddress, zrc20Denom))
+
+	// DO NOT REMOVE THE FOLLOWING CODE
+	// This section is commented until a following PR introduces the ability to withdraw delegator rewards.
+	// This validator checks will be used then to complete the whole e2e.
+
+	// res, err := r.DistributionClient.ValidatorDistributionInfo(
+	// 	r.Ctx,
+	// 	&distributiontypes.QueryValidatorDistributionInfoRequest{
+	// 		ValidatorAddress: validators[0].OperatorAddress,
+	// 	},
+	// )
+	// require.NoError(r, err)
+	// fmt.Printf("Validator 0 distribution info: %+v\n", res)
+
+	// res2, err := r.DistributionClient.ValidatorOutstandingRewards(r.Ctx, &distributiontypes.QueryValidatorOutstandingRewardsRequest{
+	// 	ValidatorAddress: validators[0].OperatorAddress,
+	// })
+	// require.NoError(r, err)
+	// fmt.Printf("Validator 0 outstanding rewards: %+v\n", res2)
+
+	// res3, err := r.DistributionClient.ValidatorCommission(r.Ctx, &distributiontypes.QueryValidatorCommissionRequest{
+	// 	ValidatorAddress: validators[0].OperatorAddress,
+	// })
+	// require.NoError(r, err)
+	// fmt.Printf("Validator 0 commission: %+v\n", res3)
+
+	// // Validator 1
+	// res, err = r.DistributionClient.ValidatorDistributionInfo(
+	// 	r.Ctx,
+	// 	&distributiontypes.QueryValidatorDistributionInfoRequest{
+	// 		ValidatorAddress: validators[1].OperatorAddress,
+	// 	},
+	// )
+	// require.NoError(r, err)
+	// fmt.Printf("Validator 1 distribution info: %+v\n", res)
+
+	// res2, err = r.DistributionClient.ValidatorOutstandingRewards(r.Ctx, &distributiontypes.QueryValidatorOutstandingRewardsRequest{
+	// 	ValidatorAddress: validators[1].OperatorAddress,
+	// })
+	// require.NoError(r, err)
+	// fmt.Printf("Validator 1 outstanding rewards: %+v\n", res2)
+
+	// res3, err = r.DistributionClient.ValidatorCommission(r.Ctx, &distributiontypes.QueryValidatorCommissionRequest{
+	// 	ValidatorAddress: validators[1].OperatorAddress,
+	// })
+	// require.NoError(r, err)
+	// fmt.Printf("Validator 1 commission: %+v\n", res3)
+}
+
+func checkCosmosBalance(r *runner.E2ERunner, address types.AccAddress, denom string) *big.Int {
```

Comment:
Can we add comment for function?

Reply:
Added in https://github.com/zeta-chain/node/commit/06d0c5d5a8a2d7ce54f84a666c7bef7784963ccf